File: net/Link.Message.Client/content/FileMessageContent.cs
using System;
using Link.Message.Client.utils;
using Newtonsoft.Json;

namespace Link.Message.Client.content
{
	

	/// <summary>
	/// 文件消息内容
	/// </summary>
	public class FileMessageContent : MessageContent
	{
        [JsonProperty("file_name")]
		public string FileName;
		public int    Size;
        [JsonProperty("download_url")]
		public string DownloadUrl;
		public string Extension;

		public FileMessageContent()
		{

		}

		public FileMessageContent(string downloadUrl, int size, string fileName, string extension) : this(downloadUrl, size, fileName + "." + extension)
		{
		}

		public FileMessageContent(string downloadUrl, int size, string fullFileName)
		{
			Guard.GuardReqiredString(downloadUrl, "downloadUrl must be set value.");
			Guard.GuardReqiredString(fullFileName, "fullFileName must be set value.");
			Guard.GuardReqiredObject(size, "size must be set value.");

			DownloadUrl = downloadUrl;
			Size = size;

			int position = fullFileName.LastIndexOf(".", StringComparison.Ordinal);
			Extension = position <= 0 ? null : fullFileName.Trim().Substring(position);
			FileName = fullFileName.Trim().Substring(0, position);

			Guard.GuardReqiredObject(size, "fullFileName must include extension name.");
			Guard.GuardReqiredObject(size, "fullFileName must include file name.");
		}
	}

}

File: java/link-message-client/src/main/java/link/message/client/content/FileMessageContent.java
package link.message.client.content;

import link.message.client.utils.Guard;

import com.alibaba.fastjson.annotation.JSONField;

/**
 * 文件消息内容
 * @author zhongt
 *
 */
public class FileMessageContent extends MessageContent {
	@JSONField(name="file_name")
	protected String fileName;
	protected int    size;
	@JSONField(name="download_url")
	protected String downloadUrl;
	protected String extension;

	public FileMessageContent() {
		
	}
	
	public FileMessageContent(String downloadUrl, int size, String fileName, String extension) {
		this(downloadUrl, size, fileName.concat(".").concat(extension));
	}
	
	public FileMessageContent(String downloadUrl, int size, String fullFileName) {
		Guard.guardReqiredString(downloadUrl,  "downloadUrl must be set value.");
		Guard.guardReqiredString(fullFileName, "fullFileName must be set value.");
		Guard.guardReqiredObject(size,         "size must be set value.");
		
		this.downloadUrl = downloadUrl;
		this.size        = size;
		
		int position = fullFileName.lastIndexOf(".");
		extension = position <=0 ? null : fullFileName.trim().substring(position);
		fileName  = fullFileName.trim().substring(0, position); 
		
		Guard.guardReqiredObject(size, "fullFileName must include extension name.");
		Guard.guardReqiredObject(size, "fullFileName must include file name.");
	}
	
	public String getFileName() {
		return fileName;
	}

	public void setFileName(String fileName) {
		this.fileName = fileName;
	}

	public int getSize() {
		return size;
	}

	public void setSize(int size) {
		this.size = size;
	}

	public String getDownloadUrl() {
		return downloadUrl;
	}

	public void setDownloadUrl(String downloadUrl) {
		this.downloadUrl = downloadUrl;
	}

	public String getExtension() {
		return extension;
	}

	public void setExtension(String extension) {
		this.extension = extension;
	}
}
